src/librustc/query/mod.rs

```diff
@@ -460,7 +461,13 @@ rustc_queries! {
 
         /// Results of evaluating const items or constants embedded in
         /// other items (such as enum variant explicit discriminants).
-        query const_eval(key: ty::ParamEnvAnd<'tcx, GlobalId<'tcx>>)
+        ///
+        /// In contrast to `const_eval_raw` this performs some validation on the constant, and
+        /// returns a proper constant that is usable by the rest of the compiler.
+        ///
+        /// **Do not use this** directly, use one of the following wrappers: `tcx.const_eval_poly`,
+        /// `tcx.const_eval_resolve`, `tcx.const_eval_instance`, or `tcx.const_eval_promoted`.
+        query const_eval_internal(key: ty::ParamEnvAnd<'tcx, GlobalId<'tcx>>)
```

Comment:
If they key difference to `_raw` is the validation, maybe it should be called `const_eval_validated`.

The current names are not great because seeing `const_eval_raw` and `const_eval_internal`, it is entirely unclear which one is higher-level.

Reply:
Or maybe we could make it `const_eval_internal_raw` and `const_eval_internal_validated` or so.

Reply:
I have changed it to `const_eval_validated`, as `const_eval_raw` is also used from `miri`, so naming it `const_eval_internal_raw` doesn't seem appropiate.